src/Message/ExtendedResponse.php

```diff
@@ -476,38 +541,87 @@ public function getCustomParameter($name)
-     * one for each subscription charge), so this function selects the correct
-     * one.
+     * When you fetch an order, it may contain multiple invoices (for example, one for each subscription charge), so
+     * this function selects the correct one.
+     *
+     * Invoices can also contain incidents (refunds, chargebacks) in the response. BlueSnap refers to these as reversal
+     * types. This is currently the only known way of retrieving refunds/chargebacks from a transaction response.
      *
+     * @param string|null $event_type
      * @return SimpleXMLElement|null
      */
-    protected function getInvoice()
+    protected function getInvoice($event_type = null)
     {
-        if ($this->invoice) {
+        if ($this->invoice !== null) {
             return $this->invoice;
         }
 
-        $this->invoice = null;
-        // find the invoice that was reqeusted
+        $invoice_match = null;
+
+        // find the invoice that was requested
         $request = $this->getRequest();
-        $transactionReference = $request->getTransactionReference();
+        $transaction_reference = $request->getTransactionReference();
 
         if ($this->data instanceof SimpleXMLElement && isset($this->data->{'post-sale-info'}->invoices)) {
-            /**
-             * @var SimpleXMLElement
-             */
+            /** @var SimpleXMLElement */
             $invoices = $this->data->{'post-sale-info'}->invoices;
-            /**
-             * @var SimpleXMLElement
-             */
+            $filter = $this->getInvoiceFilterForEventType($event_type);
+
+            /** @var SimpleXMLElement */
             foreach ($invoices->children() as $invoice) {
-                if (strval($invoice->{'invoice-id'}) === $transactionReference) {
-                    $this->invoice = $invoice;
+                if ($filter($invoice, $transaction_reference)) {
+                    $invoice_match = $invoice;
                     break;
                 }
             }
         }
-        return $this->invoice;
+        return $invoice_match;
+    }
+
+    /**
+     * @param string|null $reversal_type
+     * @return Closure
+     */
+    private function getInvoiceFilterForEventType($reversal_type = null)
+    {
+        $fn = null;
+        if ($reversal_type === Constants::REVERSAL_CHARGEBACK) {
+            $fn =
+                /**
+                 * @param SimpleXMLElement $invoice
+                 * @param string $transaction_reference
+                 * @return bool
+                 */
+                function ($invoice, $transaction_reference) {
+                    return isset($invoice->{'reversal-type'}, $invoice->{'original-invoice-id'})
+                        && (string)$invoice->{'reversal-type'} === Constants::REVERSAL_CHARGEBACK
+                        && (string)$invoice->{'original-invoice-id'} === $transaction_reference;
+                };
+        } elseif ($reversal_type === Constants::REVERSAL_REFUND) {
```

Comment:
You could combine this `if` and `elseif` by returning the same function with a closure using `$reversal_type` instead of the hardcoded constant if `$reversal_type` is not null